modules/configuration/OverallStateCollectorConfig.py

```diff
@@ -0,0 +1,15 @@
+import os
+import logging
+
+logger = logging.getLogger('esxi')
+
+
+class OverallStateCollectorConfig:
+
+    def __init__(self):
+        try:
+            self.vcenter_username = os.environ['VCENTER_USER']
+            self.vcenter_password = os.environ['VCENTER_PASSWORD']
+        except KeyError as ex:
+            logger.critical('Missing environment variable: %s' % str(ex))
+            raise SystemExit() from ex
```

Comment:
This part is used in both of your collectors, hence in both CollectorConfig classes. How about putting it into an `__init__` function of `BaseCollector` and running `super.__init__()` in the Collector itself to get the values read. Removes code duplication.

Than this CollectorConfig class can go. The other CollectorConfig class is valid and still has its use case then

Reply:
vcenter credentials are only used by overall_state_collector. Why share them bitween all collectors?